### PR TITLE
add workflow/action to auto-release buildpacks

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

Part of https://github.com/paketo-buildpacks/github-config/issues/925
- Adds a workflow to the implementation buildpacks, to release latest drafts every Wednesday morning.
- Adds an action that can be used to release just the latest draft, or all draft releases

I have gone with the cron job implementation here, instead of running this workflow everytime there is a draft release for a few reasons:
- Maintainers can expect a uniform release schedule. This helps with mental overhead and ensures if there are last-minute changes to include in a release, we know exactly when the release will go out and can plan accordingly
- Changes can be grouped together in releases, rather than creating many releases for every individual change, which might be noisy

There may be some benefits from going with a release-on-changes mechanism, but for the first iteration of this process change, I think this simplest.

## Use Cases

Tested here: 
- https://github.com/sophiewigmore/php-dist/actions/runs/9370411291/job/25797305780 (tested with two draft releases on repo, the highest version was published as expected)
- https://github.com/sophiewigmore/php-dist/actions/runs/9370318868/job/25796991533 (tested with NO draft releases on repo, no action was taken)

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X]  I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: implementation/.github/workflows/publish-releases.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+concurrency:
+  group: publish-release
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Publish Top Draft Release
```
Comment (robdimsdale) [inline implementation/.github/workflows/publish-releases.yml:16]: When we say: "top" do we mean: "top according to github", "highest semver", "chronologically most recent", or something else?

We could either rename this job or we could add a comment

**Review by robdimsdale — APPROVED** [review]

File: implementation/.github/workflows/publish-releases.yml (hunk 1 above)
Comment (sophiewigmore) [reply implementation/.github/workflows/publish-releases.yml:16]: It should be "latest"

Comment (sophiewigmore) [reply implementation/.github/workflows/publish-releases.yml:16]: or chron. most recent

Comment (sophiewigmore) [reply implementation/.github/workflows/publish-releases.yml:16]: sorry, highest semver

Comment (sophiewigmore) [reply implementation/.github/workflows/publish-releases.yml]: fixed

**Review by ForestEckhardt — APPROVED** [review]
This looks good to me @robdimsdale. Does this work for you?

**Review by robdimsdale — APPROVED** [review]

**Pull request merged**